pkg/asset/agent/joiner/clusterinfo.go

```diff
@@ -269,6 +277,45 @@ func (ci *ClusterInfo) retrieveOsImage() error {
 	return nil
 }
 
+func (ci *ClusterInfo) retrieveIgnitionEndpointWorker() error {
```

Comment:
Could we add some code comments or documentation explaining the purpose of setting the ignition endpoint? I think it is not obvious from just reading the code. It should explain that for day2 the insecured port was being used. In order to use the secured port, we needed to set a pointer ignition in assisted-service and provide a certificate to enable TLS.

Reply:
Done